sentry/src/main/java/io/sentry/SentryClient.java

```diff
@@ -139,6 +139,7 @@ private boolean shouldApplyScopeData(final @NotNull CheckIn event, final @NotNul
       }
     }
 
+    // TODO [HSM] EventProcessors from options are always executed after those from scopes
```

Comment:
Should we change how we apply event processors to combine all relevant event processors from options and scopes, sort them and then apply them?

Reply:
Good question, this is how it worked even before HSM. Do we have a good reason to change it? How much of an impact would that be for our users?

Reply:
The option to order event processors wasn't there before. Since we now have it, it should be fairly easy to just sort all of them. We can postpone this until someone actually asks for it though.